### PR TITLE
Javadoc : Explain explicitly about using Emitter interface methods synchronously

### PR DESCRIPTION
**Pull request by punitda**
Add note in Javadoc explicitly explaining about using Emitter interface methods synchronously.

- Add notes for `generate` methods in Observable and Flowable.
- Plus, add the similar note to Emitter interface.

Resolves: #6299

### CODE REVIEW

File: src/main/java/io/reactivex/Emitter.java [hunk 1]
```diff
@@ -17,6 +17,11 @@
 /**
  * Base interface for emitting signals in a push-fashion in various generator-like source
  * operators (create, generate).
+ * <p>
+ * Note that the {@link Emitter#onNext}, {@link Emitter#onError} and
+ * {@link Emitter#onComplete} methods provided to the function should be called synchronously,
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Emitter.java]: "methods provided to the function **via the `{@link Emitter}` instance** should be"

File: src/main/java/io/reactivex/Flowable.java [hunk 2]
```diff
@@ -2210,6 +2210,11 @@ public static int bufferSize() {
 
     /**
      * Returns a cold, synchronous, stateless and backpressure-aware generator of values.
+     * <p>
+     * Note that the {@link Emitter#onNext}, {@link Emitter#onError} and
+     * {@link Emitter#onComplete} methods provided to the function should be called synchronously,
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Flowable.java]: "methods provided to the function **via the `{@link Emitter}` instance** should be"

File: src/main/java/io/reactivex/Flowable.java [hunk 3]
```diff
@@ -2210,6 +2210,11 @@ public static int bufferSize() {
...
+     * never concurrently and only while the method is executing. Calling them from multiple threads
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Flowable.java]: "while the **function body** is executing."

**Review by akarnokd — CHANGES_REQUESTED** [review]
Thanks, could you rephrase all the sentences based on my suggestions in the comments?

File: src/main/java/io/reactivex/Flowable.java (hunk 2 above)
Comment (punitda) [reply src/main/java/io/reactivex/Flowable.java]: Done.

File: src/main/java/io/reactivex/Flowable.java (hunk 3 above)
Comment (punitda) [reply src/main/java/io/reactivex/Flowable.java]: Done.

File: src/main/java/io/reactivex/Emitter.java (hunk 1 above)
Comment (punitda) [reply src/main/java/io/reactivex/Emitter.java]: Done.

**Review by akarnokd — APPROVED** [review]

**Review by vanniktech — APPROVED** [review]
